Aztec/Classes/Private/Libxml2/Data/LeafNode.swift

```diff
@@ -0,0 +1,10 @@
+import Foundation
+
+/// Represents any node that can be a leaf in the DOM tree.
+///
+protocol LeafNode {
+    
+    /// Returns the text representation of the node.
+    ///
+    func text() -> String
```

Comment:
PS: when I say `Node` relies on `func text() -> String` I mean it provides that method as something its subclasses should override.  By changing `text` to become a property I start getting name collisions due to the two different expectations from `Node` and `LeafNode`.

As I said - not very clean.  But maybe we can postpone fixing that a bit?